### ISSUE TITLE
Error on Startup: Docker hv-sock proxy (vsudd) is not reachable

### ISSUE THREAD

**Issue by pmcvtm** [docker/for-win#1532]
### Expected behavior

Docker starts normally (Linux containers)

### Actual behavior

**An error occurred**
```
Docker hv-sock proxy (vsudd) is not reachable
   at Docker.Backend.ContainerEngine.Linux.ConnectToVsud(TaskCompletionSource`1 vmId) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 293
   at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 260
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 130
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145
```

### Information

  - Diagnostic ID: 2A0DC350-1622-4184-854C-1121F2C500D8/2018-01-11_09-58-11
  - Docker Version 17.12.0-ce-win46 (15048) [Stable channel]
  - Windows 10 Pro Version 10.0.16299 Build 16299   (on VMWare fusion 8.5.9)
  - Log attached: [log.txt](https://github.com/docker/for-win/files/1623260/log.txt)

### Steps to reproduce the behavior

  1. Start up Docker (automatically on Windows start or manually)
  2. Wait... and get error.

This looks like a common error message and have tried many of the solutions outlined in the other issues: switching to Windows containers and back, running as admin, deleting Moby file.

Previously have had no issues with nested virtualization... not sure if it's a nasty combo of Windows Update + Docker Version or what.

**Comment by tiggi7** [comment]
I had the same issue after upgrading to 17.12 and "enabling container"
Mi Diagnostic ID: 09CB3BBA-C407-4F82-8328-7395BA4B8718/2018-01-11_23-10-53

**Comment by Addisco** [comment]
I have the same issue, but only with linux containers. 
Switching to the edge channel solved it for now. You'll get version 17.12.0-ce-win45.
See #1539

**Comment by Gong-Bao-Chicken** [comment]
i can confirm that switching to the edge channel worked for me as well. I received the same error.

**Comment by thinusn** [comment]
I still get the error even when using `17.12.0-ce-win45 edge` , downloading `17.09.1-ce-win42 (14687)` however works for me and does not produce the error.

**Comment by awidjaja** [comment]
switching to edge channel used to work but not anymore.

**Comment by stq** [comment]
1. Stop docker service
2. Stop docker (kill process if needed)
3. Run docker as administrator

**Comment by pmcvtm** [comment]
Unfortunately for me, both the edge channel, or rolling back further to `17.09.1-ce-win42 (14687)` don't get me rolling again. It looks like they might stuck on Windows containers after trying to swap back and forth as a fix.

@stq as I said in the original issue description, running Docker as an admin hasn't worked for me-- both after stopping, or from the get-go run after rebooting.

I've ended up reverting to a snapshot of my VM before both Windows and Docker were updated, but it sounds like this is still a problem?

**Comment by winston01** [comment]
I got the same error, even after rolling back to 17.06:

```
Docker hv-sock proxy (vsudd) is not reachable
   at Docker.Backend.ContainerEngine.Linux.ConnectToVsud(TaskCompletionSource`1 vmId) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 267
   at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 251
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 123
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145
```

Information

    * Diagnostic ID: 3C01A8C2-3672-4169-B30E-4131BEF53E65/2018-01-23_13-25-42
    * Docker Version 17.06.2-ce-win27 (13194)
    * OS: Windows 10 Pro Id: 1703 Build: 15063
    * Log attached: [log.txt](https://github.com/docker/for-win/files/1655843/log.txt)

**Comment by Zelgadis87** [comment]
Tried switching to edge channel, but still get the same error:
`Docker hv-sock proxy (vsudd) is not reachable`
Diagnostic ID: 0D83DF92-BBC1-48EE-8768-D4F0B4099CE9/2018-01-24_20-23-36

**Comment by isometriq** [comment]
https://github.com/docker/for-win/issues/606
Run as admin OR disable Settings: Daemon: Experimental features

**Comment by mihaigd** [comment]
I am getting the same error.

Docker hv-sock proxy (vsudd) is not reachable

Diagnostic uploaded: CB61225F-5A27-405E-928B-27234D4AA935/2018-01-30_13-38-04

**Comment by isometriq** [comment]
Uh oh, just discovered that Android Emulator (Intel HAXM) can't run at the same time as Hyper-V...

**Comment by mihaigd** [comment]
Finally I managed to get docker to start.

This is what I did:
1. Uninstalled docker for Win 10 stable
2. Downloaded docker for Win 10 Edge
3. Installed docker Edge
4. Started docker - received the same error: Docker hv-sock proxy (vsudd) is not reachable
5. In the error reporting window I select "Reset to factory settings". This also started docker successfully.
I have no idea why after installation docker did not start, but after the reset it started.
Note: I tried the reset function repeatedly with the Stable version, but this did not solve the problem.
![docker](https://user-images.githubusercontent.com/14830500/35596379-7bca7a22-05ce-11e8-982d-032f95b4e044.jpg)

**Comment by awidjaja** [comment]
Amazing that docker team ignored this issues that has started since last year. 

I have given up trying to use docker for windows on my pc, it's a crap product with bad support and caused weeks of lost productive time. 

It's a myth that docker can run on any platform.

**Comment by ianrussel** [comment]
@awidjaja I agree, its a myth

**Comment by gregmajor** [comment]
For me, switching to:

Version 18.02.0-ce-win52 (15732)
Channel: edge
92c7ac6

...with this version of Windows:

Windows 10 Pro
Version 1709
OS Build 16299.125

...and starting Docker as admin has it working... for now.

**Comment by k20human** [comment]
Same problem here ... I switch to edge channel and it works again (for now ...)

**Comment by srini85** [comment]
Ive been getting these errors aswell. I just ended up stopping the docker windows service and starting it again. I do have experimental features turned of. It has happened a few times and restart of the service has fixed it for me

**Comment by gregmajor** [comment]
Switching to edge only worked for a short time. Now it no longer works again. Same error. I've been forced to abandon Docker for Windows, unfortunately.

**Comment by toddlucas** [comment]
I just did a fresh install of Windows 10 Pro Version 1709 (OS Build 16299.248). I figured that I would try to install Docker for Windows to see if this issue was related to my cluttered system. 

Nope. Same problem.

Uninstalling and reinstalling Docker Community Edition 17.09.0-ce-win33 2017-10-06 (13620) Stable worked.

I am astonished that this issue persists and has not even been addressed by someone at Docker. Can someone at least let us know that it's been worked on and when (or if) we can expect a resolution?

**Comment by gregmajor** [comment]
@toddlucas did you reboot a couple of times? Uninstalling and re-installing made it work for me... until I rebooted.

**Comment by toddlucas** [comment]
Actually, I didn't. I will give that a try when I have time. Thanks!

**Comment by licor** [comment]
Turn off the UAC solved the issue.

**Comment by arthurdn** [comment]
**Removing or changing** of priority for third party software with **high CPU consumption** fixed everything.

**Comment by fernandolozer** [comment]
Is this a Windows configuration? or a docker configuration? @arthurdn 
> Removing or changing of priority for third party software with high CPU consumption fixed everything.

**Comment by arthurdn** [comment]
@fernandolozer it is a Windows. I have 'Folding at home', just stopped CPU calculations on it, GPU calculations still runs with docker in parallel. ALL (!!!) other comments in similar topics did not helped completely.

**Comment by simonm3** [comment]
I have this problem too. resetting docker worked but deleted all my images and containers.....and a week later I am back with exactly the same error.

**Comment by jdmcnair** [comment]
Current edge version (18.02.0-ce) works for me.

**Comment by guillemsola** [comment]
I have this message everyday in my computer. Doing a reset doesn't help as the same errors pumps up after.

Docker with win10 in my enterprise laptop has also connectivity issues but this is a blocking issue. Overall experience not positive.

**Comment by talbx** [comment]
I have the same problem everyday on version 17.09.0-ce-win33. 
Usually killing Docker.Service and DockerForWindows in the Task Manager and restart as Admin is solving this issue.

**Comment by simonm3** [comment]
> Usually killing Docker.Service and DockerForWindows in the Task Manager
and restart as Admin is solving this issue.

That worked for me too thanks.

**Comment by sjwright1986** [comment]
This issue is preventing me from working efficiently. I've tried resetting docker, launching docker as admin, reinstalled windows, it very very frustrating. Come on Docker, at least let us know if you've acknowledged the issue but it'll be some time to rectify? Any information is better than none.

At this rate I think I'll go back to using vagrant

**Comment by rtillet** [comment]
I spent a whole day to try to make it work. After reading this thread, my conclusion is that "Docker for Windows" is a JOKE.

**Comment by techcto** [comment]
+1

**Comment by halofx** [comment]
I ran up against this issue after the 1709 update, and Docker Version 17.12.0-ce-win47 (15139). I have tried all the mentioned fixes:
- Running as Admin.
- Moving the location Hyper-V stores VMs.
-Changing permission.
-Reinstalling Docker.
-Reinstalling Hyper-V.

When I switch to Windows Containers (which I am not very familiar with) Docker launches, and says it is running, but no VM is created in Hyper-V. Is this normal?

When launching as Linux Containers, the VM gets created, starts and runs for 60 seconds, then shutdown. It this rebooots again, runs 60 seconds a 2nd time, then shuts down and displays the hv-sock error.

**Comment by halofx** [comment]
Finally found some success by completely uninstalling Docker with this script, https://success.docker.com/article/How_to_completely_remove_Docker_in_Windows_10

And then rolling back to Docker 17.06.2-ce-win27, and not updating to the current version.

**Comment by gregmajor** [comment]
I tried those exact steps @halofx but unfortunately it only worked until I rebooted. I also discovered that the "completely remove Docker" script doesn't even remotely understand the word "completely". There were folders and registry entries littered all over.

**Comment by srini85** [comment]
Just out of curiosity, how many people here had this issue with Docker for Windows as their first Docker installation? I had Docker toolbox which I uninstalled then went to Docker for Windows. Wondering if it's the sequence of events and dangling files/services that's causing it.

**Comment by toddlucas** [comment]
For me, I had been using DfW for a long time. Generally upgrading whenever prompted. Then this happened. I was able to work around the issue by uninstalling and then reinstalling an earlier version.

As I mentioned before, I recently wiped my machine and did a clean install of W10 Pro so I thought I would try again with the latest. I got the same issue and worked around it using the same uninstall, then downgrade reinstall that fixed it the first time.

I haven't used Docker Toolbox and I didn't have Virtual Box or anything on this machine. Just Hyper-V.

**Comment by halofx** [comment]
@gregmajor  Fingers crossed, It has survived a couple reboots.

@srini85 I had previously been using Docker for Windows, but it fell apart after the 1709 Fall Creators update. Which I then reinstalled as part of that process with the current version. Unfortunately, I have no idea which version of Docker I had been using previously before 1709.

**Comment by licor** [comment]
Once again this bastard issue came back to frustrating me 👎

**Comment by zbennett10** [comment]
Like others have said, the quick fix for me was:
1. Stop the docker service.
2. Quit docker.
3. Run docker as an admin.

I thoroughly enjoy using Docker, but Docker for Windows definitely has some issues that need to be worked on.

**Comment by gregmajor** [comment]
Interestingly enough, if I perform all the steps that @halofx outlined _combined_ with stopping the Docker service and re-starting as Administrator, I can get it running.

Looking through all the reported issues, it's clear that Docker for Windows is a long way from being ready for prime-time. I'm glad that the community is sticking together and offering up what has worked for them so that we might all benefit.

**Comment by abokov** [comment]
Have same issue, running as admin doesn't really help. What works for me ( not really what I want, but it makes docker running at least ) is to switch to windows containers.

**Comment by ctmcisco** [comment]
For anyone following this ticket:

Try the following:

1. Switch or stay on Windows Containers.
2. Disable the "Start when Windows starts" setting.


![image](https://user-images.githubusercontent.com/14852041/37053107-e9a96bd8-2140-11e8-9bdf-b264980b6127.png)

3. Exit Docker application
4. restart docker services via PowerShell

get-service *docker* | Restart-Service -Force -Verbose

5. Launch Docker , Run as Admin
6. Swtich to Linux Containers.

TEST:

Reboot Windows 10
Launch Docker 
Does it work?

You can also parse the docker log in PowerShell for troubleshooting:

cat C:\Users\USERPROFILE\AppData\Local\Docker\log.txt -tail 10 -wait

**Comment by rn** [comment]
@pmcvtm we do not support running Docker for Windows inside a Windows VM. See https://docs.docker.com/docker-for-windows/troubleshoot/#running-docker-for-windows-in-nested-virtualization-scenarios
therefore closing

**Comment by cwe1ss** [comment]
This also happens on regular windows installations.

**Comment by toddlucas** [comment]
@rn Thank you for taking a look at this. We have been waiting a long time for someone at Docker to give it some attention. I would like to point out that this is not about running DfW in a VM. I believe most of us are running vanilla Windows 10 Pro (thus Hyper-V). This is a real issue that is causing people to abandon Docker since it's unusable for them.

**Comment by simonm3** [comment]
Closing this issue doesn't make it go away. The whole point of docker is enabling cross platform environments. If docker does not work on windows then it is useless.

**Comment by rn** [comment]
I closed the issue because of the original post was running on VMware, which is not supported (and documented as such).

There are many causes for the VM not starting (which then results in `Docker hv-sock proxy (vsudd) is not reachable` messages). I understand that people see the same *symptoms* on different systems but the *cause* is likely very different. Running on VMware is potentially one of them and we don't support it because it requires the VMware VM to be configure carefully and we have no visibility of it.

**Comment by simonm3** [comment]
OK but several of us have exactly the same problem running just on windows.
Should we post a new issue?

**Comment by further-reading** [comment]
@simonm3 Yup! As they said there could be different causes so a new issue would help if you can't find any open issues about this error on non-VM windows.

Note that the "run as admin" workaround some people mention worked for me just now.

**Comment by olegsorokin** [comment]
I changed the **Docker for Windows Service** startup type to manual. That helped.

**Comment by beylkhanovdamir** [comment]
my current workaround of this issue is sequential restart of Docker service process and then of Docker itself. Docker should be running under Administrator

**Comment by khellinx** [comment]
Wow ... reaction of @rn was like: 'works on my machine' ... meh.

**Comment by jamestharpe** [comment]
Fresh install of Windows + Fresh install of D4W = this issue :-(

**Comment by ArcaneEngineer** [comment]
This happened as I had insufficient disk space on C:\ to run the instance - as soon as I freed up from only 220Mb to 1.2Gb, the instance started fine. Also needed to run Docker as administrator on Windows 10.

So Docker really needs to get some idea of the size of the instance and inform user accordingly on startup attempt.

**Comment by ThinWonton** [comment]
spending my lots of time for trying to run docker on windows...
always show me a boring tips 'docker is starting '

**Comment by kiahmed** [comment]
just quit docker gracefully and restart it works often
